nwb_conversion_tools/utils/neo.py

```diff
@@ -0,0 +1,570 @@
+import warnings
+
+import neo.io.baseio
+import pynwb
+
+from .conversion_tools import add_devices
+
+PathType = Union[str, Path, None]
+
+
+response_classes = {
+    "voltage_clamp": pynwb.icephys.VoltageClampSeries,
+    "current_clamp": pynwb.icephys.CurrentClampSeries,
+    "izero": pynwb.icephys.IZeroClampSeries,
+}
+
+stim_classes = {
+    "voltage_clamp": pynwb.icephys.VoltageClampStimulusSeries,
+    "current_clamp": pynwb.icephys.CurrentClampStimulusSeries,
+}
+
+
+# TODO - get electrodes metadata
+def get_electrodes_metadata(neo_reader, electrodes_ids: list, block: int = 0) -> list:
+    """
+    Get electrodes metadata from Neo reader. The typical information we look for is the information
+    accepted by pynwb.icephys.IntracellularElectrode:
+    - name – the name of this electrode
+    - device – the device that was used to record from this electrode
+    - description – Recording description, description of electrode (e.g., whole-cell, sharp, etc) COMMENT: Free-form text (can be from Methods)
+    - slice – Information about slice used for recording.
+    - seal – Information about seal used for recording.
+    - location – Area, layer, comments on estimation, stereotaxis coordinates (if in vivo, etc).
+    - resistance – Electrode resistance COMMENT: unit: Ohm.
+    - filtering – Electrode specific filtering.
+    - initial_access_resistance – Initial access resistance.
+
+    Args:
+        neo_reader ([type]): Neo reader
+        electrodes_ids (list): List of electrodes ids.
+        block (int, optional): Block id. Defaults to 0.
+
+    Returns:
+        list: List of dictionaries containing electrodes metadata
+    """
+    return []
+
+
+# Get number of electrodes
+def get_number_of_electrodes(neo_reader) -> int:
+    """
+    Get number of electrodes from Neo reader
+
+    Args:
+        neo_reader ([type]): Neo reader
+
+    Returns:
+        int: number of electrodes
+    """
+    return len(neo_reader.header["signal_channels"])
+
+
+# Get number of segments
+def get_number_of_segments(neo_reader, block: int = 0) -> int:
+    """
+    Get number of segments from Neo reader
+
+    Args:
+        neo_reader ([type]): Neo reader
+        block (int, optional): Block id. Defaults to 0.
+
+    Returns:
+        int: number of electrodes
+    """
+    return neo_reader.header["nb_segment"][block]
+
+
+# Get command traces (e.g. voltage clamp command traces)
+def get_command_traces(neo_reader, block: int = 0, segment: int = 0, cmd_channel: int = 0) -> Tuple[list, str, str]:
+    """
+    Get command traces (e.g. voltage clamp command traces).
+
+    Args:
+        neo_reader ([type]): [description]
+        block (int, optional): [description]. Defaults to 0.
+        segment (int, optional): [description]. Defaults to 0.
+        cmd_channel (int, optional): ABF command channel (0 to 7). Defaults to 0.
+
+    Returns:
+        list: [description]
+    """
+    traces, titles, units = neo_reader.read_raw_protocol()
```

Comment:
This is not generic to neo but only for AxonIO case

Reply:
for now AxonIO is the only Icephys we're dealing with, we can expand this for other cases when they appear